### PR TITLE
Fix reconnect error

### PR DESCRIPTION
**Pull request by aneeshkp**
Signed-off-by: Aneesh Puttur <aneeshputtur@gmail.com>
This PR fixes three items
1. On delete and create of  event of pt4l config , the sidecar was cleaning up threshold values and resetting to default 100 nano secs. 
This was briefly setting offset check on  default values.
Also on delete there was an error trying to close already closed channel.
2. when  application losses connection to amqp and then reconnects , the QDR  thread monitoring for reconnects and when trying to build the AMQ object again , it loses the Override Function that was set earlier.Thsi PR fixes to retain  those values
3. Updated SDK AMQ and  go-SDK to the latest

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -77,6 +78,12 @@ func main() {
 	prometheus.Unregister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}))
 	prometheus.Unregister(collectors.NewGoCollector())
 
+	nodeIP := os.Getenv("NODE_IP")
+	if nodeIP != "" {
+		amqpHost = strings.Replace(amqpHost, "NODE_IP", nodeIP, 1)
```
Comment (jzding) [inline cmd/main.go:83]: amqpHost is default to `amqp:localhost:5672`. Should the `NODE_IP` be `localhost` here?
```
amqpHost = strings.Replace(amqpHost, "localhost", nodeIP, 1)
```

Comment (aneeshkp) [reply cmd/main.go:83]: this is  an option to use NODE_IP if required , the user will  set the transport host  value as amqp://NODE_IP:30005 and  the container will read  from env  replace NODE_IP to the actual node ip value 
```
  - name: NODE_IP
              valueFrom:
                fieldRef:
                  fieldPath: status.hostIP
```

Comment (aneeshkp) [reply cmd/main.go:83]: only replaces the  string NODE_IP with  the ip address

Comment (jzding) [reply cmd/main.go:83]: got it.

**Review by jzding — APPROVED** [review]

**Pull request merged**